neqo-transport/src/connection/mod.rs

```diff
@@ -1937,6 +1937,9 @@ impl Connection {
             };
 
             let path = close.path().borrow();
+            if path.is_temporary() {
+                return Err(Error::InternalError);
+            }
```

Comment:
This is a good fix.  Can you add a comment here about how we might get here?

```suggestion
            // In some error cases, we will not be able to make a new, permanent path.
            // For example, if we run out of connection IDs and the error results from
            // a packet on a new path, we avoid sending (and the privacy risk) rather
            // than reuse a connection ID.
            if path.is_temporary() {
                return Err(Error::InternalError);
            }
```